<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="diffusion_planner_param_path" default="$(find-pkg-share autoware_diffusion_planner)/config/diffusion_planner.param.yaml"/>

  <node pkg="autoware_diffusion_planner" exec="autoware_diffusion_planner_node" name="diffusion_planner_node" output="screen">
    <param from="$(var diffusion_planner_param_path)" allow_substs="true"/>
    <remap from="~/output/trajectory" to="/planning/diffusion_planner/trajectory"/>
    <remap from="~/output/trajectories" to="/planning/generator/diffusion_planner/candidate_trajectories"/>
    <remap from="~/output/predicted_objects" to="~/diffusion_planner/predicted_objects"/>
    <remap from="~/input/odometry" to="/localization/kinematic_state"/>
    <remap from="~/input/acceleration" to="/localization/acceleration"/>
    <remap from="~/input/route" to="/planning/mission_planning/route"/>
    <remap from="~/input/traffic_signals" to="/perception/traffic_light_recognition/traffic_signals"/>
    <remap from="~/input/tracked_objects" to="/perception/object_recognition/tracking/objects"/>
    <remap from="~/input/vector_map" to="/map/vector_map"/>
    <param name="onnx_model_path" value="$(var onnx_model_path)"/>
    <param name="args_path" value="$(var args_path)"/>
    <param name="build_only" value="$(var build_only)"/>
  </node>
</launch>
